### ISSUE TITLE
Context length setting is ignored with ollama

### ISSUE THREAD

**Issue by swarmer** [BoltAI/BoltAI#156]
**Describe the bug**
At least with gemma2-27b, context length setting in Ollama model settings appeared to do nothing.
To get larger context length, I had to create an ollama modelfile with `PARAMETER num_ctx 8192`, create my custom model, and use that. I don't know if this is specific to gemma2-27b or some general issue with ollama, I suspect the latter since num_ctx is the same parameter for all models.

**To Reproduce**

1. Select gemma2-27b in ollama.
2. Set context length to something like 8192.
3. Ask the model something.
4. grep ollama logs for `n_ctx`, It will say 2048. Or ask about the beginning of a large prompt.

**Expected behavior**
8192 context length is used.

**Version (please complete the following information):**
 - macOS: 14.6.1
 - app version v1.19.2
 - ollama 0.3.8

**Comment by longseespace** [comment]
This is expected as the Ollama integration in BoltAI is via the OpenAI-compatible API server, not via direct API. I might add support for this in the future.

**Comment by swarmer** [comment]
Is there any reason not to hide the context length option, since it gives false hope? Does it do anything in some scenario?

**Comment by longseespace** [comment]
Yeah I will hide it until it's implemented. In the past I used this to show a warning if user's message is too long (larger than the context length) but since the tokenizer was not accurate, I removed it.

**Comment by dbpprt** [comment]
Any timeline for the ollama API to support longer context sizes?

**Comment by longseespace** [comment]
> Any timeline for the ollama API to support longer context sizes?

It depends on Ollama and not BoltAI. I can opt to use Ollama directly but then we won’t be able to use Plugins or other OpenAI standardized parameters. 

See: https://github.com/ollama/ollama/issues/2722

**Comment by longseespace** [comment]
Also see this guide: https://github.com/ollama/ollama/blob/main/docs/openai.md#setting-the-context-size
